src/shared/kbd-util.c

```diff
@@ -31,52 +31,48 @@ static int nftw_cb(
                 return 0;
 
         p = strdup(basename(fpath));
-        if (!p)
-                return FTW_STOP;
+        if (!p) {
+                errno = ENOMEM;
+                return -1;
+        }
 
         e = endswith(p, ".map");
-        if (e)
-                *e = 0;
-
-        e = endswith(p, ".map.gz");
+        if (!e)
+                e = endswith(p, ".map.gz");
```

Comment:
nit: this change is unrelated to this commit, and is reworked anyway in the last commit, wouldn't it be cleaner to avoid it?

Reply:
The optimization in the last commit didn't initially occur to me. If it's OK, I would rather not rework this.

Reply:
sure no problem, it's just a minor thing and rebases are a pain